src/daft-logical-plan/src/optimization/rules/reorder_joins/join_graph.rs

```diff
@@ -46,36 +51,76 @@ impl Display for JoinNode {
     }
 }
 
-/// JoinEdges currently represent a bidirectional edge between two relations that have
-/// an equi-join condition between each other.
-#[derive(Debug)]
-struct JoinEdge(JoinNode, JoinNode);
+#[derive(Clone, Debug)]
+pub(crate) struct JoinCondition {
+    pub left_on: String,
+    pub right_on: String,
+}
 
-impl JoinEdge {
-    /// Helper function that summarizes join edge information.
-    fn simple_repr(&self) -> String {
-        format!("{} <-> {}", self.0, self.1)
+impl JoinCondition {
+    pub(crate) fn flip(&self) -> Self {
+        JoinCondition {
+            left_on: self.right_on.clone(),
+            right_on: self.left_on.clone(),
+        }
     }
 }
 
-impl Display for JoinEdge {
+pub(crate) struct JoinAdjList(
+    pub HashMap<LogicalPlanRef, HashMap<LogicalPlanRef, Vec<JoinCondition>>>,
```

Comment:
I would strong avoid using the `LogicalPlanRef` as a hash key here. Whenever you do and insert or a look up the logical plan has to be hashed and then equity has to be computed. I would instead just assign each Node an incrementing ID or use the `Arc::ptr` as the ID

Reply:
Alternatively you can also use something like a 

```
struct GraphNodeKey {
  Id: usize,
  plan: LogicalPlanRef // ignored for hash and eq calcuations
}

```